src/python/pants/core/goals/lint.py

```diff
@@ -225,61 +228,82 @@ async def lint(
         "Iterable[type[LintTargetsRequest]]", union_membership.get(LintTargetsRequest)
     )
     fmt_target_request_types = cast("Iterable[type[FmtRequest]]", union_membership.get(FmtRequest))
-    target_request_types = [
-        *lint_target_request_types,
-        *fmt_target_request_types,
-    ]
-    file_request_types = union_membership[LintFilesRequest]
+    file_request_types = cast(
+        "Iterable[type[LintFilesRequest]]", union_membership[LintFilesRequest]
+    )
 
-    _check_ambiguous_request_names(*target_request_types, *file_request_types)
+    _check_ambiguous_request_names(
+        *lint_target_request_types, *fmt_target_request_types, *file_request_types
+    )
 
     specified_names = determine_specified_tool_names(
         "lint",
         lint_subsystem.only,
-        target_request_types,
+        [
+            *lint_target_request_types,
+            *fmt_target_request_types,
+        ],
         extra_valid_names={request.name for request in file_request_types},
     )
-    target_requests = tuple(
-        request_type(
-            request_type.field_set_type.create(target)
-            for target in targets
-            if (
-                request_type.name in specified_names
-                and request_type.field_set_type.is_applicable(target)
-            )
-        )
-        for request_type in target_request_types
-    )
 
-    def address_str(fs: FieldSet) -> str:
-        return fs.address.spec
+    def is_specified(request_type: type[StyleRequest] | type[LintFilesRequest]):
+        return request_type.name in specified_names
+
+    lint_target_request_types = filter(is_specified, lint_target_request_types)
+    fmt_target_request_types = filter(is_specified, fmt_target_request_types)
+    file_request_types = filter(is_specified, file_request_types)
 
     def batch(field_sets: Iterable[FieldSet]) -> Iterator[list[FieldSet]]:
-        return partition_sequentially(
+        partitions = partition_sequentially(
             field_sets,
-            key=address_str,
+            key=lambda fs: fs.address.spec,
             size_target=lint_subsystem.batch_size,
             size_max=4 * lint_subsystem.batch_size,
         )
+        for partition in partitions:
+            if partition:
```

Comment:
Were you observing empty partitions from `partition_sequentially`? I don't think that that was intended, so might be better to fix over there.

Reply:
Oh good point. I think I was copying over the previous `if field_sets` and got switched up here. I'll double check things

Reply:
Everything looks good if field_sets is empty. Weird it had the conditional before.